examples/decorators/uniffi.toml

```diff
@@ -0,0 +1,43 @@
+[bindings.python]
+cdylib_name = "wrapper_types"
+
+[bindings.ruby]
+cdylib_name = "wrapper_types"
+
+[bindings.swift]
+cdylib_name = "wrapper_types"
+
+[bindings.kotlin]
+cdylib_name = "wrapper_types"
+
+# Defining the log-errors decorator for the Python bindings
+[decorators.log-errors.python]
+imports = ["logging"]
+code = """
+error_logger = logging.getLogger("errors")
+
+def logErrors(func):
+  try:
+    return func()
+  except:
+    error_logger.error("Error calling {}".format(func), exc_info=True)
+    raise
+"""
+decorator = "logErrors"
+
+# Defining the io decorator for the Python bindings
+[decorators.io.kotlin]
+imports = [
+    "kotlinx.coroutines.Dispatchers",
+    "kotlinx.coroutines.CoroutineScope"
+]
+code = """
+ioScope = CoroutineScope(Dispatchers.IO)
+"""
+decorator = "ioScope.launch"
+
+# I think the return type can be inferred and we don't need the next line.  But
+# this is an example of how the return type could be specified.  Swift might
+# need something like this.
+return_type = "Deferred<{}>" # "{}" will be replaced with the decorated function's return type
```

Comment:
Kotlin seems to support type inference of return values in some cases - but let's assume it didn't and/or swift can't make it work. Let's say we had 2 functions described in UDL, one returning a String and one returning an Int - and we want to decorate them both as "io", but want to avoid using `Any` in Kotlin because that would be a step backward. I wonder if there's a way to just assume the "correct" signatures exist? Obviously you'd need to arrange <handwave> for there to be 2 functions with different return values (and therefore, almost certainly different names). If you later go ahead and mark a function returning (say) a float, you can imagine us generating the code that naively assumes a float version of the decorator exists and let the compiler complain if it doesn't.

This isn't that different to the uniffi contract with your Rust code today - it just blindly assumes that what you wrote matches what actually exists, and if not, the compiler complains.

Reply:
As long as the decorator is a generic function, I think it all works out.  The generated code would be something like this:

```
fun foo(a: Int, b: String) = ioScope.launch {
   {{ func|lift }} (
      {{ func|ffi_func_name }}({{ func.args.a|lower }}, {{ func.args.b|lower}})
   )
}
```